include/fields2cover/swath_generator/swath_generator_base.h

```diff
@@ -14,10 +14,12 @@
 
 namespace f2c::sg {
 
+enum class SwathOverlapType {NO_OVERLAP, END_OVERLAP, MIDDLE_OVERLAP, EVENLY_DISTRIBUTED_OVERLAP};
```

Comment:
Add int to each type, so it'll prevent future errors if someone use the indexs to refer to the SwathOverlapType and we remove any of the methods. So something like:

```
enum class SwathOverlapType {
  NO_OVERLAP = 0,
  END_OVERLAP = 1,
  MIDDLE_OVERLAP = 2,
  EVENLY_DISTRIBUTED_OVERLAP = 3,
};
```